### PR TITLE
CVA6: Infrastructure improvements

### PR DESCRIPTION
**Pull request by MarioOpenHWGroup**
Some improvements in the current infrastructure:
- Auto detect `CV_SW_PREFIX` if not set.
- Set `RISCV_GCC` and `RISCV_OBJCOPY` if not set.
- Avoid return statements when fatal errors.
- Detection of the `tohost` addr now has more compatibility across systems.
- Other improvements.

### CODE REVIEW

File: cva6/regress/install-cva6.sh
Comment (cathales) [inline cva6/regress/install-cva6.sh:80]:
```suggestion
  git -C core-v-cores/cva6 checkout $CVA6_HASH;
```

`--git-dir` is to use the current repository, but using another `.git/` directory. Here you want to act on another repository so you want option `-C` instead.

File: cva6/regress/install-cva6.sh
Comment (MarioOpenHWGroup) [reply cva6/regress/install-cva6.sh:43]: if the source is not done, the following statement on line 47 will fail, as it is not preserved from the `install-verilator` script.

File: cva6/regress/install-cva6.sh
Comment (MarioOpenHWGroup) [reply cva6/regress/install-cva6.sh:80]: My bad, I will correct it :smile:

File: cva6/regress/install-verilator.sh
Comment (MarioOpenHWGroup) [reply cva6/regress/install-verilator.sh:12]: The objective of the change was to standardize the script. There are some places of the scripts where it is used instead of `-z`. To be honest I prefer too the other option so if it is fine for you then I would like to change all the places where `! [ -n`  is used to `-z`

**Comment by MarioOpenHWGroup** [comment]
@cathales if you could take another look it would be great :smile:

**Comment by MikeOpenHWGroup** [comment]
Hey @MarioOpenHWGroup and @cathales, there are multiple "conversations" in this PR (which is a _good_ thing).  I believe some of them are now fixed, but it is hard to tell.  Please use the "Resolve conversation" button to close discussions that are resolved.  Thanks.

**Comment by MarioOpenHWGroup** [comment]
> Hey @MarioOpenHWGroup and @cathales, there are multiple "conversations" in this PR (which is a _good_ thing). I believe some of them are now fixed, but it is hard to tell. Please use the "Resolve conversation" button to close discussions that are resolved. Thanks.

For sure, I have resolved all the issues in the thread :+1:

**Comment by MarioOpenHWGroup** [comment]
@cathales a final revision would be appreciated :smiley:

**Comment by cathales** [comment]
@MikeOpenHWGroup I wanted to mark conversations as resolved but I am not allowed to do so.

PR LGTM, I think it would be great if @ASintzoff could have a look too, as he is one of the original authors.

**Comment by MikeOpenHWGroup** [comment]
> @MikeOpenHWGroup I wanted to mark conversations as resolved but I am not allowed to do so.

Yes, this is a strange GitHub feature: `You can resolve a conversation in a pull request if you opened the pull request or if you have write access to the repository where the pull request was opened`.  In this context, "write access" is a someone who can approve a pull-request (a Committer).  I think this is broken.

**Comment by ASintzoff** [comment]
LGTM

**Review by MikeOpenHWGroup — APPROVED** [review]

**Pull request merged**
